ipatests/test_integration/test_ntp_options.py

```diff
@@ -0,0 +1,268 @@
+
+            args3 = ['ipa-client-install', '-N',
+                     '--ntp-pool=%s' % self.ntp_pool]
+            client_install = self.client.run_command(args3, raiseonerr=False)
+            assert client_install.returncode == 2
+            assert exp_pool_str in client_install.stderr_text
+
+        finally:
+            self.cleanup()
+
+    def test_replica_promotion_with_ntp_options(self):
+        """
+        test to verify that replica promotion with ntp --ntp-server,
+        --ntp-pool and -N or --no-ntp option would fail
+        """
+        exp_str = "NTP configuration cannot be updated during promotion"
+
+        try:
+            tasks.install_master(self.master, setup_dns=False)
+            tasks.install_client(self.master, self.replica)
+
+            replica_install = self.replica.run_command(
+                ['ipa-replica-install', '--no-ntp'],
+                raiseonerr=False)
+            assert replica_install.returncode == 1
+            assert exp_str in replica_install.stderr_text
+
+            replica_install = self.replica.run_command(
+                ['ipa-replica-install', '--ntp-server=%s' % self.ntp_server1],
+                raiseonerr=False)
+            assert replica_install.returncode == 1
+            assert exp_str in replica_install.stderr_text
+
+            replica_install = self.replica.run_command(
+                ['ipa-replica-install', '--ntp-pool=%s' % self.ntp_pool],
+                raiseonerr=False)
+            assert replica_install.returncode == 1
+            assert exp_str in replica_install.stderr_text
+
+        finally:
+            self.cleanup()
+
+    def test_replica_promotion_without_ntp(self):
+        """
+        test to verify that replica promotion without ntp options
+        - ipa-client-install with ntp option
+        - ipa-replica-install without ntp option
+        will be successful
+        """
+        exp_str = "ipa-replica-install command was successful"
+        expected_msg = "Configuration of chrony was changed by installer."
+        ntp_args = ['--ntp-pool=%s' % self.ntp_pool]
+
+        try:
+            server_install = tasks.install_master(self.master, setup_dns=False,
+                                                  extra_args=ntp_args)
+            assert expected_msg in server_install.stderr_text
+
+            client_install = tasks.install_client(self.master, self.replica,
+                                                  extra_args=ntp_args)
+            assert expected_msg in client_install.stderr_text
+
+            replica_install = tasks.install_replica(self.master, self.replica,
+                                                    promote=False)
+            assert exp_str in replica_install.stderr_text
+
+            cmd = self.replica.run_command(['cat', paths.CHRONY_CONF])
+            assert self.ntp_pool in cmd.stdout_text
+
+        finally:
+            self.cleanup()
+
+    def cleanup(self):
```

Comment:
Doesn't `teardown_method` work on this branch?

Reply:
Hi @netoarmando, I still wait for results of TMP commit on #2638. After that i will propagate changes here :)